### DISCUSSION TITLE
Jump to definition in java libraries

### DISCUSSION THREAD

**Discussion by antosha417** [scalameta/nvim-metals#375] [Q&A] [answered]
Hi! 
Is it possible to `gd` from scala to java code using metals? 
This seems not to work for me. 
When I do `:lua vim.lsp.buf.definition()` on some java code that I use in scala project nothing happens.
There is no errors in `metals.log`

**Comment by ckipp01** [comment, marked as answer]
Hey @antosha417! Yes, this should totally work. Is this some methods in a Scala file, or are you actually in a Java file? If you're in a Java file and looking for some navigation you'll want to make sure you have `java` in the filetype autocmd that starts Metals. Like:

```lua
  api.nvim_create_autocmd("FileType", {
    pattern = { "scala", "sbt", "java" },
    callback = function()
      require("metals").initialize_or_attach(metals_config)
    end,
    group = nvim_metals_group,
  })
```

Do you have that? Also, could you take a look at the `:MetalsRunDoctor`? Do you see anything funky in there?

**Reply by antosha417** [reply]
It turns out I did have `java` in the pattern. So metals was not attached.
